client/websocket_client.go

```diff
@@ -0,0 +1,193 @@
+// (at your option) any later version.
+
+// Eris-RT is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with Eris-RT.  If not, see <http://www.gnu.org/licenses/>.
+
+package client
+
+import (
+	"bytes"
+	"fmt"
+	"time"
+
+	"github.com/tendermint/go-rpc/client"
+	"github.com/tendermint/go-wire"
+
+	log "github.com/eris-ltd/eris-logger"
+
+	"github.com/eris-ltd/eris-db/txs"
+	ctypes "github.com/eris-ltd/eris-db/rpc/tendermint/core/types"
+)
+
+const (
+	MaxCommitWaitTimeSeconds = 10
+)
+
+type Confirmation struct {
+	BlockHash []byte
+	Event     txs.EventData
+	Exception error
+	Error     error
+}
+
+// NOTE [ben] Compiler check to ensure ErisNodeClient successfully implements
+// eris-db/client.NodeClient
+var _ NodeWebsocketClient = (*ErisNodeWebsocketClient)(nil)
+
+type ErisNodeWebsocketClient struct {
+	// TODO: assert no memory leak on closing with open websocket
+	tendermintWebsocket *rpcclient.WSClient
+}
+
+// Subscribe to an eventid
+func (erisNodeWebsocketClient *ErisNodeWebsocketClient) Subscribe(eventid string) error {
+	// TODO we can in the background listen to the subscription id and remember it to ease unsubscribing later.
+	return erisNodeWebsocketClient.tendermintWebsocket.Subscribe(eventid)
+}
+
+// Unsubscribe from an eventid
+func (erisNodeWebsocketClient *ErisNodeWebsocketClient) Unsubscribe(subscriptionId string) error {
+	return erisNodeWebsocketClient.tendermintWebsocket.Unsubscribe(subscriptionId)
+}
+
+// Returns a channel that will receive a confirmation with a result or the exception that
+// has been confirmed; or an error is returned and the confirmation channel is nil.
+func (erisNodeWebsocketClient *ErisNodeWebsocketClient) WaitForConfirmation(tx txs.Tx, chainId string, inputAddr []byte) (chan Confirmation, error) {
+	// check no errors are reported on the websocket 
+	if err := erisNodeWebsocketClient.assertNoErrors(); err != nil {
+		return nil, err
+	}
+
+	// Setup the confirmation channel to be returned
+	confirmationChannel := make(chan Confirmation, 1)
+	var latestBlockHash []byte
+
+	eid := txs.EventStringAccInput(inputAddr)
+
+	// Read the incoming events
+	go func() {
+		var err error
+		for {
+			resultBytes := <- erisNodeWebsocketClient.tendermintWebsocket.ResultsCh
+			result := new(ctypes.ErisDBResult)
+			if wire.ReadJSONPtr(result, resultBytes, &err); err != nil {
+				// keep calm and carry on
+				log.Errorf("eris-client - Failed to unmarshal json bytes for websocket event: %s", err)
+				continue
+			}
+			
+			event, ok := (*result).(*ctypes.ResultEvent)
+			if !ok {
+				// keep calm and carry on
+				log.Error("eris-client - Failed to cast to ResultEvent for websocket event")
```

Comment:
yes, but it should only be `EventDataTx` in the case of the `InputAddr` as event string